daq_main2.py

```diff
@@ -35,6 +35,8 @@
 logger.addHandler(handler1)
 logger.addHandler(handler2)
 
+perform_server_checks()
+setBlConfig("visitDirectory", os.getcwd())
```

Comment:
It is a bit opaque to me that after `perform_sever_checks` that we are sure that the cwd is a valid visit directory.  It might be better to have a helper function that looks like

```python
def get_visit_directory():
    candidate = os.getcwd()
    if is_valid(candidate):
        raise ...
    return candidate

```

I think this would better ensure that the check does not get accidentally lost and reduces the number of places `os.getcwd()` exists in the code.

Reply:
I think I see what you are getting at with the other Slack conversation - as the visit directory gets changed as the server is kept running for multiple user visits, how can we ensure that it is always pointing to a good visit directory?

the medium term plan is to separate LSDC server from its cwd so that it will always write to the visit directory. the other useful (to local contact) output from the way the server is run is the log file. we will plan on using an environment variable to set where the logging output will be.

Reply:
This is the only place in the codebase that `setBlConfig("visitDirectory", ...)` is called, how can it be changed?

I was more worried that at some point in the future `perform_server_checks` would stop checking `os.getcwd()` and we would lose control of it again.  A function that does the validation and return the valid path (or raises) would be more robust (and more testable!) it the medium term.  It would also easily allow for validating setting this later.

Reply:
as part of the changes to separate the LSDC server from cwd as mentioned above, we could implement this suggestion when we make the server persist across multiple visits, at which point the check would become important as a server check won't necessarily be performed between different visits.